Code/Framework/AzCore/AzCore/Debug/Trace.h

```diff
@@ -12,6 +12,10 @@
 #include <AzCore/O3DEKernelConfiguration.h>
 #include <cstdarg>
 
+#if defined(AZ_PLATFORM_LINUX)
+    #include <cstdio>
+#endif
```

Comment:
Using platform specific macros should be avoided: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#code.

We prefer to extract platform specific functionality into separate files.

That being said we don't need to conditionally add this include for linux here.
We should unconditionally add this include to [AzCore/base.h](https://github.com/o3de/o3de/blob/4abdbbdde5884ed5d1729362745f805e5373e9be/Code/Framework/AzCore/AzCore/base.h#L46-L53)